src/PSRule.Rules.Azure/rules/Azure.Storage.Rule.ps1

```diff
@@ -88,6 +88,13 @@ Rule 'Azure.Storage.ContainerSoftDelete' -Ref 'AZR-000289' -Type 'Microsoft.Stor
     }
 }
 
+# Synopsis: Enable Malware Scanning in Microsoft Defender for Storage.
+Rule 'Azure.Storage.DefenderCloud.MalwareScan' -Ref 'AZR-000384' -Type 'Microsoft.Storage/storageAccounts' -Tag @{ release = 'Preview'; ruleSet = '2023_06'; 'Azure.WAF/pillar' = 'Security'; } -Labels @{ 'Azure.MCSB.v1/control' = 'DP-2', 'LT-1' } {
+    $malwareConfigured = @(GetSubResources -ResourceType 'Microsoft.Security/DefenderForStorageSettings' |
+        Where-Object { $_.properties.malwareScanning.onUpload.isEnabled -eq $True })
+    $Assert.GreaterOrEqual($malwareConfigured, '.', 1).Reason($LocalizedData.ResStorageMalwareScanning, $PSRule.TargetName)
```

Comment:
We want to find cases when the malware scanning has been disabled. Try the following instead.

```suggestion
    $malwareDisabled = @(GetSubResources -ResourceType 'Microsoft.Security/DefenderForStorageSettings' |
        Where-Object { $_.properties.malwareScanning.onUpload.isEnabled -eq $False })
    $Assert.Count($malwareDisabled, '.', 0).Reason($LocalizedData.ResStorageMalwareScanning, $PSRule.TargetName)
```

Additionally, I think we need to a pre-filter to skip the rule if `properties.publicNetworkAccess` is set to `Disabled` which is not supported as mentioned.